Plugins/ActSVG/include/Acts/Plugins/ActSVG/IndexedSurfacesSvgConverter.hpp

```diff
@@ -0,0 +1,375 @@
+    for (unsigned int ib0 = 1u; ib0 <= axes[0u]->getNBins(); ++ib0) {
+      for (unsigned int ib1 = 1u; ib1 <= axes[1u]->getNBins(); ++ib1) {
+        typename index_grid::grid_type::index_t lbin;
+        lbin[0u] = ib0;
+        lbin[1u] = ib1;
+        highlightIndices.push_back(indexGrid.grid.atLocalBins(lbin));
+        // Register the bin naming
+        pGrid._bin_ids.push_back(std::string("- bin : [") +
+                                 std::to_string(ib0) + std::string(", ") +
+                                 std::to_string(ib1) + std::string("]"));
+        if (estimateR) {
+          pGrid._reference_r = constrain.medium(binR);
+        }
+      }
+    }
+  }
+  return std::tie(pSurfaces, pGrid, highlightIndices);
+}
+
+}  // namespace IndexedSurfacesConverter
+
+namespace View {
+
+/// Convert into an acts::svg::object with an XY view
+///
+/// @param pIndexGrid is the proto object
+/// @param identification is the to be translated id_ for actsvg
+///
+/// @return an svg object that can be written out directly
+static inline actsvg::svg::object xy(const ProtoIndexedSurfaceGrid& pIndexGrid,
+                                     const std::string& identification) {
+  auto [pSurfaces, pGrid, pIndices] = pIndexGrid;
+
+  // Create the master object
+  actsvg::svg::object xyIndexedGrid;
+  xyIndexedGrid._id = identification;
+  xyIndexedGrid._tag = "g";
+
+  // The surfaces
+  std::vector<actsvg::svg::object> sObs;
+  for (const auto& s : pSurfaces) {
+    if (pGrid._type == actsvg::proto::grid::e_z_phi) {
+      sObs.push_back(Acts::Svg::View::zrphi(s, s._name));
+    } else {
+      sObs.push_back(Acts::Svg::View::xy(s, s._name));
+    }
+  }
+
+  // The grid
+  auto gOb =
+      actsvg::display::grid(identification + std::string("_grid"), pGrid);
+
+  // The connectors
+  actsvg::connectors::connect_action(gOb._sub_objects, sObs, pIndices);
+
+  // Add them all
+  xyIndexedGrid.add_objects(sObs);
+
+  auto xmax = xyIndexedGrid._x_range[1u];
+  // The assoication info boxes
+  for (auto [ig, gTile] : enumerate(gOb._sub_objects)) {
+    // Target surface text
+    std::vector<std::string> binText;
+    binText.push_back("Source:");
+    binText.push_back(pGrid._bin_ids[ig]);
+    binText.push_back("Target:");
+    for (auto& sis : pIndices[ig]) {
+      binText.push_back(std::string("- object: ") + std::to_string(sis));
+    }
+    // Make the connected text
+    auto cText = actsvg::draw::connected_text(
+        "bla", {static_cast<actsvg::scalar>(1.1 * xmax), 0}, binText,
+        actsvg::style::font{}, actsvg::style::transform{}, gTile);
+    xyIndexedGrid.add_object(cText);
+  }
+
+  xyIndexedGrid.add_object(gOb);
+  // Return the grid
+  return xyIndexedGrid;
+}
+
+/// Fake zphi method, delegate to xy
+///
+/// @param pIndexGrid is the proto object
+/// @param identification is the to be translated id_ for actsvg
+///
+/// @return an svg object that can be written out directly
+static inline actsvg::svg::object zphi(
```

Comment:
Added:

```
/// @note this works, because the actual display at the end is 2D
/// in an x-y plane and the parameters are appropriatly defined
///
```